macros/misc/randomNamesPronouns.pl

```diff
@@ -0,0 +1,525 @@
+	'Ravi'      => 'he',
+	'Ray'       => 'they',
+	'Rebecca'   => 'she',
+	'Renee'     => 'she',
+	'Rita'      => 'she',
+	'Ronda'     => 'she',
+	'Ross'      => 'he',
+	'Ryan'      => 'he',
+	'Samantha'  => 'she',
+	'Sarah'     => 'she',
+	'Scot'      => 'he',
+	'Sean'      => 'he',
+	'Sebastian' => 'he',
+	'Selena'    => 'she',
+	'Shane'     => 'he',
+	'Sharell'   => 'she',
+	'Sharnell'  => 'she',
+	'Sherial'   => 'she',
+	'Stephanie' => 'she',
+	'Stephen'   => 'he',
+	'Subin'     => 'she',
+	'Sydney'    => 'she',
+	'Tammy'     => 'she',
+	'Teresa'    => 'she',
+	'Thanh'     => 'he',
+	'Tien'      => 'he',
+	'Tiffany'   => 'she',
+	'Timothy'   => 'he',
+	'Tracey'    => 'she',
+	'Virginia'  => 'she',
+	'Wendy'     => 'she',
+	'Wenwu'     => 'he',
+	'Will'      => 'he'
+};
+
+@last_names = qw(
+	Adams   Allen     Alvarez    Anderson Bailey  Baker    Bennet    Brooks
+	Brown   Campbell  Carter     Castillo Chavez  Clark    Collins   Cook
+	Cooper  Cox       Cruz       Davis    Diaz    Edwards  Evans     Flores
+	Foster  Garcia    Gomez      Gonzales Gray    Green    Gutierrez Hall
+	Harris  Hernandez Hill       Howard   Hughes  Jackson  James     Jimenez
+	Johnson Jones     Kelly      Kim      King    Lee      Lewis     Long
+	Lopez   Martin    Martinez   Mendoza  Miller  Mitchell Moore     Morales
+	Morgan  Morris    Murphy     Myers    Nelson  Nguyen   Ortiz     Parker
+	Patel   Perez     Peterson   Phillips Pina    Price    Ramirez   Ramos
+	Reed    Reyes     Richardson Rivera   Roberts Robinson Rodriguez Rogers
+	Ross    Ruiz      Sanchez    Sanders  Scott   Smith    Stewart   Sullivan
+	Taylor  Thomas    Thompson   Torres   Turner  Walker   Ward      Watson
+	White   Williams  Wilson     Wood     Wright  Young
+);
+
+=head2 randomPerson
+
+Returns a person as a Person object from a list in the macro.
+
+=head3 Examples and options
+
+=over
+
+=item * No arguments returns a single random person
+
+Example
+
+  randomPerson()
+
+=item * C<n =E<gt> k> returns an array of k Person objects with unique names.
+
+Example
+
+  randomPerson(n=>5)
+
+returns an array of 5 Person objects with unique names.
+
+=item * C<names =E<gt> arrayref> returns an array of specified Person objects.
+
+If the arrayref is in the form of C<[['name1','pronoun1'],['name2','pronoun2'],...]> then
+an array of Person objects of the names and pronouns specified.
+
+Example:
+
+  my ($a,$b,$c) = randomPerson(names => [['Bart','he'], ['Lisa','she'], ['Matty','they']]);
+
+If the arrayref is just in the form of C<['name1','name2',...]> then the names are specified
+and the pronouns are randomly generated.
+
+  my ($p1, $p2, $p3) = randomPerson( names => ['Larry', 'Moe', 'Curly']);
+
+=back
+
+=cut
+
+sub randomPerson {
+	my %options = (
+		n => 1,
+		@_
+	);
+
+	# if the names are passed in.
+	return
+		map { Person->new(name => $_->[0] // $_, pronoun => $_->[1] // list_random('he', 'she', 'they')) }
+		@{ $options{names} }
+		if ($options{names});
+
+	if ($options{n} == 1) {
```

Comment:
If n is 1 (or assumed to be 1) then it enters this block of the conditional code, and the keys of the hash are not being sorted before a random selection.

I think this will be moot once you settle on how to store the big list of names as an array ref.

Reply:
Good catch.  I'll rethink a better way to store the names.